### PR TITLE
New example for Modules

### PR DESCRIPTION
**Pull request by cmelchior**
Depends on #1156  

This adds a new example (realmModuleExample) consisting of two modules showing how Realm can be used with library projects using Realm Modules. It also shows how to setup ProGuard.

### CODE REVIEW

File: examples/realmModuleAppExample/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+
+    <string name="app_name">RealmModule example</string>
+    <string name="hello_world">Hello world!</string>
```
Comment (bmunkholm) [inline examples/realmModuleAppExample/src/main/res/values/strings.xml]: hello world ?

**Comment by cmelchior** [comment]
Ready for review: @emanuelez @nhachicha @kneth

File: examples/realmModuleExample/app/src/main/java/io/realm/examples/appmodules/ModulesExampleActivity.java [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    private LinearLayout rootLayout = null;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_modules_example);
+        rootLayout = ((LinearLayout) findViewById(R.id.container));
+        rootLayout.removeAllViews();
+
+        // The default Realm instance implicitly know about all classes in the realmModuleAppExample Android Studio
```
Comment (kneth) [inline examples/realmModuleExample/app/src/main/java/io/realm/examples/appmodules/ModulesExampleActivity.java]: "know" -> "knows"

File: examples/realmModuleExample/app/src/main/java/io/realm/examples/appmodules/ModulesExampleActivity.java [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+        showStatus("Copy objects between Realms");
+        showStatus("Number of pigs on the farm : " + farmRealm.where(Pig.class).count());
+        showStatus("Copy pig from defaultRealm to farmRealm");
+        Pig defaultPig = defaultRealm.where(Pig.class).findFirst();
+        farmRealm.beginTransaction();
+        farmRealm.copyToRealm(defaultPig);
+        farmRealm.commitTransaction();
+        showStatus("Number of pigs on the farm : " + farmRealm.where(Pig.class).count());
+
+        // Each realm is restricted to only accept the classes in their schema.
```
Comment (kneth) [inline examples/realmModuleExample/app/src/main/java/io/realm/examples/appmodules/ModulesExampleActivity.java]: "realm" -> "Realm"

**Comment by kneth** [comment]
:+1: besides my comments.

**Comment by emanuelez** [comment]
Why are all the template files been moved to an app subfolder?

**Comment by cmelchior** [comment]
Because that way the folder structure matches the folder structure in the distribution examples. This means I don't have to add exceptions to the logic copying files and it can be done 1:1. This was not needed before as the folder layout where all the same, but

**Comment by emanuelez** [comment]
:+1:

File: examples/realmModuleExample/app/src/main/java/io/realm/examples/appmodules/ModulesExampleActivity.java [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+import io.realm.examples.librarymodules.model.Dog;
+import io.realm.examples.librarymodules.model.Elephant;
+import io.realm.examples.librarymodules.model.Lion;
+import io.realm.examples.librarymodules.model.Zebra;
+import io.realm.examples.librarymodules.modules.DomesticAnimalsModule;
+import io.realm.examples.librarymodules.modules.ZooAnimalsModule;
+import io.realm.exceptions.RealmException;
+
+/**
+* This example demonstrates how you can use modules to control which objects belong to which Realms and how you can
```
Comment (bmunkholm) [inline examples/realmModuleExample/app/src/main/java/io/realm/examples/appmodules/ModulesExampleActivity.java]: objects -> classes

**Pull request merged**